internal/handlers/common/update.go

```diff
@@ -328,6 +334,45 @@ func processIncFieldExpression(doc *types.Document, updateV any) (bool, error) {
 	return changed, nil
 }
 
+// processMaxFieldExpression changes document according to $max operator.
+// If the document was changed it returns true.
+func processMaxFieldExpression(doc *types.Document, updateV any) (bool, error) {
+	maxExpression := updateV.(*types.Document)
+
+	var changed bool
+
+	for _, field := range maxExpression.Keys() {
+		val, _ := doc.Get(field)
+
+		maxVal, err := maxExpression.Get(field)
+		if err != nil {
+			// if max field does not exist, don't change anything
+			continue
+		}
+
+		// if the key is not found, set it
+		if val != nil {
+			res := types.CompareOrder(val, maxVal, types.Ascending)
+			switch res {
+			case types.Equal:
+				fallthrough
+			case types.Greater:
+				continue
+			case types.Less:
+			case types.Incomparable:
```

Comment:
Should I leave an empty `types.Less` case or make a nolint here?

Reply:
You could add a comment like “document value less than update one, set the new value”